src/mount/mount-tool.c

```diff
@@ -1157,8 +1182,10 @@ int main(int argc, char* argv[]) {
         if (streq_ptr(arg_mount_options, "defaults"))
                 arg_mount_options = mfree(arg_mount_options);
 
-        if (!is_device_path(arg_mount_what))
```

Comment:
The problem is that if it is a mount path then `arg_mount_what` will never be set to anything but `NULL`